pkg/cache/cache.go

```diff
@@ -443,13 +446,22 @@ func (c *Cache) UpdateClusterQueue(cq *kueue.ClusterQueue) error {
 func (c *Cache) DeleteClusterQueue(cq *kueue.ClusterQueue) {
 	c.Lock()
 	defer c.Unlock()
+	ctx := context.TODO()
+	logger := ctrl.LoggerFrom(ctx)
 	curCq := c.hm.ClusterQueue(cq.Name)
 	if curCq == nil {
 		return
 	}
-	if features.Enabled(features.LocalQueueMetrics) {
-		for _, q := range c.hm.ClusterQueue(cq.Name).localQueues {
-			metrics.ClearLocalQueueCacheMetrics(metrics.LQRefFromLocalQueueKey(q.key))
+	if features.Enabled(features.LocalQueueMetrics) && metrics.GetLocalQueueMetrics().Enabled {
+		for _, q := range curCq.localQueues {
+			lqSlice := strings.Split(q.key, "/")
```

Comment:
```suggestion
			lqRef := metrics.LQRefFromLocalQueueKey(q.key)
```

I think we can reuse this.